km52/Antiushyn_Yaroslav/laboratory1.sql

```diff
@@ -1,2 +1,136 @@
--- LABORATORY WORK 1
--- BY Antiushyn_Yaroslav
+/*---------------------------------------------------------------------------
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+вставляти дані до таблиць
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+/*
+CREATE User Antiyshin IDENTIFIED by Antiyshin
+DEFAULT TABLESPACE "USERS"
+TEMPORARY TABLESPACE "TEMP";
+ALTER USER Antiyshin QUOTA 100M ON USERS:
+GRANT "CONNECT" to Antiyshin;
+INSERT INTO to USERS; 
+
+*/
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+Класна кімната має парти та стільці.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+CREATE User Antiyshin IDENTIFIED by Antiyshin
+DEFAULT TABLESPACE "USERS"
+TEMPORARY TABLESPACE "TEMP";
+ALTER USER Antiyshin QUOTA 100M ON USERS:
+
+CREATE TABLE Desk (desk_id varchar(30) not null);
+ALTER TABLE Desk 
+  ADD CONSTRAINT desk_id_pk PRIMARY KEY (desk_id);
+  
+CREATE TABLE Chair (chair_id varchar(30) not null);
+ALTER TABLE Desk 
+  ADD CONSTRAINT chair_id_pk  PRIMARY KEY (chair_id);
+
+CREATE TABLE Classroom (
+  desk_id_fk varchar(30) not null
+  chair_id_fk varchar(30) not null
+);
+
+ALTER TABLE Classroom 
+  ADD CONSTRAINT desk_id_pk PRIMARY KEY (desk_id_fk,chair_id_fk);
+  
+Alter
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+  
+/* --------------------------------------------------------------------------- 
+3. Надати додаткові права користувачеві (створеному у пункті № 1) для створення таблиць, 
+внесення даних у таблиці та виконання вибірок використовуючи команду ALTER/GRANT. 
+Згенерувати базу даних використовуючи код з теки OracleScript та виконати запити: 
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+3.a. 
+Скільком покупцям продано найдорожчий товар?
+Виконати завдання в SQL. 
+4 бали
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+3.b. 
+Визначити скільки різних електронних адрес зберігається в таблиці CUSTOMERS - назвавши це поле count_email.
+Виконати завдання в SQL. 
+4 бали
+
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+c. 
+Вивести ім’я та пошту покупця, як єдине поле client_name, для тих покупців, що не мають замовлення.
+Виконати завдання в алгебрі Кодда. 
+4 бали
+
```

Comment:
```
SELECT DISTINCT
  TRIM(CUSTOMERS.cust_name)
  || ' '
  || TRIM(CUSTOMERS.cust_email)
  AS "client_name"
FROM CUSTOMERS
  WHERE cust_id NOT IN (SELECT cust_id FROM ORDERS);
```

Reply:
Rename (
Project (
Customers Where cust_id NOT IN Project (Orders) cust_id)  Trim(cust_name) CONCAT Trim(cust_email)) as "client_name"